### ISSUE TITLE
Elasticsearch 2.0.0 don't start if marvel-agent is set in plugin.mandatory

### ISSUE THREAD

**Issue by rvrignaud** [elastic/elasticsearch#14357]
Running elasticsearch 2.0.0 on ubuntu 14.04, marvel-agent is installed:
```
/usr/share/elasticsearch/bin/plugin list
Installed plugins in /usr/share/elasticsearch/plugins:
    - marvel-agent
    - license
```
However if I set marvel-agent in the list of mandatory plugins in elasticsearch.yml I get the following Exception:
```
[2015-10-29 16:53:52,138][INFO ][node                     ] [schinf-0000-prd1] version[2.0.0], pid[5239], build[de54438/2015-10-22T08:09:48Z]
[2015-10-29 16:53:52,140][INFO ][node                     ] [schinf-0000-prd1] initializing ...
[2015-10-29 16:53:52,404][ERROR][bootstrap                ] Exception
ElasticsearchException[Missing mandatory plugins [marvel-agent]]
	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:149)
	at org.elasticsearch.node.Node.<init>(Node.java:144)
	at org.elasticsearch.node.NodeBuilder.build(NodeBuilder.java:145)
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:170)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:270)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)
```

If I don't set marvel-agent in plugin.mandatory, elasticsearch start correctly and list marvel-agent as installed.

**Comment by clintongormley** [comment]
As a workaround, specify `plugin.mandatory: marvel`

**Comment by tlrx** [comment]
The offending bug is located [here](https://github.com/elastic/elasticsearch/blob/v2.0.0/core/src/main/java/org/elasticsearch/plugins/PluginsService.java#L132) where it used the name defined in the plugin's class rather than the name defined in the plugin's properties file.

This is already fixed in 2.x, 2.1 and [master](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/plugins/PluginsService.java#L129) branch of elasticsearch. We don't need to rename anything in Marvel for now.

I created #14479 to backport the fix in 2.0 too, in case 2.0.1 is released.

**Comment by clintongormley** [comment]
2.1.0 has been released. Closing
